tests/MongoDB.Driver.Tests/PinnedShardRouterTests.cs

```diff
@@ -172,11 +172,17 @@ private DisposableMongoClient CreateDisposableClient(EventCapturer eventCapturer
             var client = DriverTestConfiguration.CreateDisposableClient((MongoClientSettings settings) =>
                 {
                     settings.ClusterConfigurator = c => c.Subscribe(eventCapturer);
-                    settings.LocalThreshold = TimeSpan.FromMilliseconds(1000);
+                    settings.LocalThreshold = TimeSpan.FromMilliseconds(5000);
                 }, 
                 useMultipleShardRouters);
             var timeOut = TimeSpan.FromSeconds(60);
-            SpinWait.SpinUntil(() => client.Cluster.Description.Type != ClusterType.Unknown, timeOut).Should().BeTrue();
+
+            var cluster = client.Cluster;
+            SpinWait.SpinUntil(condition: () =>
+                cluster.Description.Type != ClusterType.Unknown
```

Comment:
Note: if all servers are connected (which is the condition I suggest checking in the one liner below) then by definition the cluster type will not be Unknown, which is why this clause can be omitted.

Reply:
I've taken your suggestion!